api/src/main/java/org/pmiops/workbench/dataset/DataSetServiceImpl.java

```diff
@@ -702,6 +702,119 @@ private String getQualifiedColumnName(Domain currentDomain, String columnName) {
 
+  @Override
+  public List<String> generateMicroarrayCohortExtractCodeCells(
+      DbWorkspace dbWorkspace,
+      String qualifier,
+      Map<String, QueryJobConfiguration> queriesByDomain) {
+    String joinedDatasetVariableNames =
+        queriesByDomain.entrySet().stream()
+            .map(
+                e ->
+                    "dataset_"
+                        + qualifier
+                        + "_"
+                        + Domain.fromValue(e.getKey()).toString().toLowerCase()
+                        + "_df")
+            .collect(Collectors.joining(", "));
+
+    final String cohortSampleNamesFilename = "cohort_sample_names_" + qualifier + ".txt";
+    final String cohortSampleMapFilename = "cohort_sample_map_" + qualifier + ".csv";
+    final String cohortVcfFilename = "cohort_" + qualifier + ".vcf";
+
+    return ImmutableList.of(
+        "person_ids = set()\n"
+            + "datasets = ["
+            + joinedDatasetVariableNames
+            + "]\n"
+            + "\n"
+            + "for dataset in datasets:\n"
+            + "    if 'PERSON_ID' in dataset:\n"
+            + "        person_ids = person_ids.union(dataset['PERSON_ID'])\n"
+            + "    elif 'person_id' in dataset:\n"
+            + "        person_ids = person_ids.union(dataset['person_id']) \n"
+            + "\n\n"
+            + "with open('"
+            + cohortSampleNamesFilename
+            + "', 'w') as cohort_file:\n"
+            + "    for person_id in person_ids:\n"
+            + "        cohort_file.write(str(person_id) + '\\n')\n"
+            + "    cohort_file.close()\n",
+        "%%bash\n\n"
+            + "uuid=$(cat /proc/sys/kernel/random/uuid | sed s/-/_/g)\n"
+            // TODO: Writing to the "tmp" dataset is a temporary workaround until an alternative,
+            // RW-5735
+            + "EXPORT_TABLE=\"fc-aou-cdr-synth-test.tmp_shared_cohort_extract.${uuid}\"\n"
+            + "\n"
+            + "python3 /genomics/microarray/raw_array_cohort_extract.py \\\n"
+            + "          --dataset fc-aou-cdr-synth-test.microarray_data \\\n"
+            + "          --fq_destination_table ${EXPORT_TABLE} \\\n"
+            + "          --query_project ${GOOGLE_PROJECT} \\\n"
+            // TODO: Replace hardcoded dataset reference: RW-5748
+            + "          --sample_mapping_table fc-aou-cdr-synth-test.microarray_data.sample_list \\\n"
+            + "          --cohort_sample_names_file "
+            + cohortSampleNamesFilename
+            + " \\\n"
+            + "          --sample_map_outfile "
+            + cohortSampleMapFilename
+            + "\n"
+            + "\n"
+            + "gatk ArrayExtractCohort \\\n"
+            + "        -R gs://fc-aou-cdr-synth-test-genomics/extract_resources/Homo_sapiens_assembly19.fasta \\\n"
+            + "        -O "
+            + cohortVcfFilename
+            + " \\\n"
+            + "        --probe-info-csv /genomics/microarray/probe_info.csv \\\n"
```

Comment:
Looking at the code - it seems we can probably skip this flag entirely - and it will read this from the table: https://github.com/broadinstitute/gatk/blob/ah_var_store/src/main/java/org/broadinstitute/hellbender/tools/variantdb/arrays/ArrayExtractCohort.java#L199-L204